### PR TITLE
Plutus pab logging: ChainIndex / SigningProcess (SCP-1687)

### PR DESCRIPTION
**Pull request by gilligan**
**Description**:
 Change `ChainIndex` and `SigningProcess` to use Tracer via `Freer.Log` (part of SCP-1687).

**Details**:
- Moved some definitions to local Type Module to resolve circular imports
- Additional minor refactorings
- Added some comments

-----

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-pab/src/Plutus/PAB/Monitoring.hs [hunk 1]
```diff
@@ -106,6 +107,14 @@ toSeverity = \case
   L.Alert     -> Alert
   L.Emergency -> Emergency
 
+handleLogEffects ::
```
Comment (gilligan) [inline plutus-pab/src/Plutus/PAB/Monitoring.hs:110]: The name maybe isn't the best? Open for suggestions.

File: plutus-pab/src/Plutus/PAB/PABLogMsg.hs [hunk 2]
```diff
@@ -46,10 +52,11 @@ data AppMsg =
     | InstalledContract Text
     | ContractInstance ContractExe [ContractInstanceId]
     | TxHistoryItem Tx
-    | ContractHistoryItem Int (ContractInstanceState ContractExe) -- index
+    | ContractHistoryItem Int (ContractInstanceState ContractExe)
```
Comment (gilligan) [inline plutus-pab/src/Plutus/PAB/PABLogMsg.hs:55]: No idea what the `--index` would even mean here so I dropped it

File: plutus-pab/app/Main.hs [hunk 3]
```diff
@@ -374,7 +383,12 @@ reportContractHistoryParser =
         (ReportContractHistory <$> contractIdParser)
         (fullDesc <> progDesc "Show the state history of a smart contract.")
 
-{- Note [Use of iohk-monitoring in PAB]
```
Comment (j-mueller) [inline plutus-pab/app/Main.hs:377]: We adopted the "Note [xxx]" and "see note [xxx]" pattern from GHC and use it all over the place in the plutus repo, I think we should stick with it

File: plutus-pab/app/Main.hs [hunk 4]
```diff
@@ -409,76 +423,121 @@ We have to use 'natTracer' in some places to turn 'Trace IO a' into
...
         threads <- traverse forkCommand commands
         putStrLn "Started all commands."
         waitAny threads
   where
-
     forkCommand ::  Command -> IO (Async ())
     forkCommand subcommand = do
       putStrLn $ "Starting: " <> show subcommand
-      -- see note [Use of iohk-monitoring in PAB]
+      -- see [note on monitoring in pab](#iohk-monitoring-pab)
```
Comment (j-mueller) [inline plutus-pab/app/Main.hs:466]:
```suggestion
      -- see note [Use of iohk-monitoring in PAB]
```

File: plutus-pab/app/Main.hs [hunk 5]
```diff
@@ -492,7 +551,7 @@ main = do
     (trace :: Trace IO AppMsg, switchboard) <- setupTrace_ logConfig pabComponentName
 
     -- 'TracerLoggerT IO' has instances for 'MonadLogger' and 'MonadUnliftIO'.
-    -- see note [Use of iohk-monitoring in PAB]
+    -- see [note on monitoring in pab](#iohk-monitoring-pab)
```
Comment (j-mueller) [inline plutus-pab/app/Main.hs:554]: see above

File: plutus-pab/src/Plutus/PAB/App.hs [hunk 6]
```diff
@@ -238,8 +238,7 @@ mkEnv Config { dbConfig
 runApp :: Trace IO PABLogMsg -> CM.Configuration -> Config -> App a -> IO (Either PABError a)
 runApp theTrace logConfig config action =
     runTraceLoggerT
-
-    -- see note [Use of iohk-monitoring in PAB]
+    -- see [note on monitoring in pab](#iohk-monitoring-pab)
```
Comment (j-mueller) [inline plutus-pab/src/Plutus/PAB/App.hs:241]: ditto

File: plutus-pab/src/Plutus/PAB/MonadLoggerBridge.hs [hunk 7]
```diff
@@ -89,7 +89,7 @@ toLogLevel = \case
     LevelOther _ -> L.Info
 
 -- | Interpret 'MonadLogger' effect using a 'Trace'.
--- see note [Use of iohk-monitoring in PAB] for why this is necessary.
+-- see [note on monitoring in pab](#Main.iohk-monitoring-pab) for why this is necessary.
```
Comment (j-mueller) [inline plutus-pab/src/Plutus/PAB/MonadLoggerBridge.hs:92]: ditto

File: plutus-pab/src/Plutus/PAB/Monitoring.hs (hunk 1 above)
Comment (j-mueller) [reply plutus-pab/src/Plutus/PAB/Monitoring.hs:110]: maybe `runLogTrace` ? I tend to use `handle` for functions that remove one or more of the effects but still return an `Eff effs`

**Review by j-mueller — CHANGES_REQUESTED** [review]
The logging stuff looks good but we should keep the notes intact.

File: plutus-pab/app/Main.hs (hunk 3 above)
Comment (michaelpj) [reply plutus-pab/app/Main.hs:377]: It's even documented! https://github.com/input-output-hk/plutus/blob/master/CONTRIBUTING.adoc#notes

Comment (gilligan) [reply plutus-pab/app/Main.hs:377]: @michaelpj yeah - i'm not gonna get out of this one easily. I think i just went "oh i can use haddock anchors" and forgot about the whole notes thing - my bad :sweat_smile:

**Review by j-mueller — APPROVED** [review]

**Pull request merged**